OneSignalSDK/onesignal/src/main/java/com/onesignal/LocationGMS.java

```diff
@@ -158,18 +191,24 @@ else if (permissionList.contains("android.permission.ACCESS_COARSE_LOCATION")) {
                      requestPermission = "android.permission.ACCESS_COARSE_LOCATION";
                }
 
-               if (requestPermission != null && promptLocation)
+               if (requestPermission != null && promptLocation) {
```

Comment:
Can you comment the cases being handled here, lots of if and else that might not be intuitive to someone who hasn't seen the codebase

Reply:
Still needs comments I think